carbon-design-system/carbon
# feat(ContentSwitcher): add icon-only ContentSwitcher

Closes https://github.com/carbon-design-system/carbon/issues/12811

Adds in a new icon-only `ContentSwitcher`

#### Changelog

**New**

- `IconSwitch` component was created, and rendered as a child of `ContentSwitcher`. When this is used instead of `Switch`, it will render an `IconButton`.

```jsx
<ContentSwitcher>
    <IconSwitch name="one" text="Table of Contents">
      <TableOfContents />
    </IconSwitch>
    <IconSwitch name="two" text="Workspace Test">
      <Workspace />
    </IconSwitch>
    <IconSwitch name="three" text="View Mode">
      <ViewMode_2 />
    </IconSwitch>
  </ContentSwitcher>
```

- The `name` passed into `IconSwitch` will render the `Tooltip` text. This will also set the `aria-label` on the `IconButton`
- Tests and snapshots added for the new `IconSwitch`
- Styles for the new icon-only `ContentSwitcher`

**Changed**

- `ContentSwitcher` now passes along `size` to its children.
- Updated the `size` prop types to valid v11 sizes.
- Adjusted styles for the new icon-only `ContentSwitcher`
- Added a `wrapperClasses` prop to `IconButton`. I needed a way to surface selected + disabled states to the outermost wrapper, and adding one in `IconSwitch` was causing the `role="tab"` be too far removed from its parent, and causing screenreader issues (Each button read `Tab 1 of 1`). Since it is now just 1 level removed, the screen reader announces the number of tabs correctly. 

**Removed**

- Removed the `with layer` snapshot and replaced it with the new icon-only story. The `with layer` story has no visual differences from the default story. 

#### Testing / Reviewing

Go to `ContentSwitcher` and check out the icon-only story. 
- Ensure it matches the spec, and tooltips appear. 
- Ensure sizes are able to change, and supports both selection modes. 
- Try disabling an icon button and ensure it works. 

Make sure there are no regressions to the standard `ContentSwitcher`
